cmd/maya-apiserver/app/server/volume_endpoint_v1alpha1.go

```diff
@@ -32,81 +30,208 @@ func (s *HTTPServer) volumeV1alpha1SpecificRequest(resp http.ResponseWriter, req
 	}
 
 	switch req.Method {
-	case "PUT", "POST":
+	case "POST":
 		return volOp.create()
 	case "GET":
-		return volOp.get()
+		return volOp.httpGet()
+	case "DELETE":
+		return volOp.httpDelete()
 	default:
 		return nil, CodedError(405, ErrInvalidMethod)
 	}
 }
 
-// TODO Move the delete out to HTTP DELETE
-//
-// get deals with HTTP GET request
-func (v *volumeAPIOpsV1alpha1) get() (interface{}, error) {
-	// Extract info from path after trimming
-	path := strings.TrimPrefix(v.req.URL.Path, "/0.6.0/volumes")
+// httpGet deals with http GET request
+func (v *volumeAPIOpsV1alpha1) httpGet() (interface{}, error) {
+	// Extract name of volume from path after trimming
+	path := strings.TrimSpace(strings.TrimPrefix(v.req.URL.Path, "/latest/volumes"))
 
-	// Is req valid ?
-	if path == v.req.URL.Path {
-		return nil, CodedError(405, ErrInvalidMethod)
+	// list cas volumes
+	if path == "/" {
+		return v.list()
 	}
 
-	switch {
+	// read a cas volume
+	volName := strings.TrimPrefix(path, "/")
+	return v.read(volName)
+}
 
-	case strings.Contains(path, "/info/"):
-		volName := strings.TrimPrefix(path, "/info/")
-		return v.read(volName)
-	case strings.Contains(path, "/delete/"):
-		volName := strings.TrimPrefix(path, "/delete/")
-		return v.delete(volName)
-	case path == "/":
-		return v.list()
-	default:
+// httpDelete deals with http DELETE request
+func (v *volumeAPIOpsV1alpha1) httpDelete() (interface{}, error) {
+	// Extract name of volume from path after trimming
+	volName := strings.TrimSpace(strings.TrimPrefix(v.req.URL.Path, "/latest/volumes/"))
+
+	// check if req url has volume name
+	if len(volName) == 0 {
 		return nil, CodedError(405, ErrInvalidMethod)
 	}
+
+	return v.delete(volName)
 }
 
-func (v *volumeAPIOpsV1alpha1) create() (interface{}, error) {
+func (v *volumeAPIOpsV1alpha1) create() (*v1alpha1.CASVolume, error) {
+	glog.Infof("cas template based volume create request was received")
 
-	glog.Infof("Received volume add request: Version: 'v1alpha1'")
+	vol := &v1alpha1.CASVolume{}
```

Comment:
Some of the methods above, like for volume list and delete we are extracting volumeName from path. But here for volume creation we are not extracting it from path rather using CASVolume struct.
Is it intentional ? I mean how do we decide from where volumeName is to be extracted.
I suspect a `mayactl cli` failure for volume creation.Please correct me if I am missing something.

Reply:
@sonasingh46 api server expects the payload to have volume structure during PUT, POST calls. When its a get or delete it expects from the URL path.

Reply:
Yes I got this one. You are right.
When we create a volume we have name in payload itself.